pages/models.py

```diff
@@ -548,64 +548,197 @@ def get_context(self, *args, **kwargs):
         ]
 
 
-
-## CANDIDATES
-
+# CANDIDATES
 class CandidateEndorsementPage(Page):
-    body = RichTextField()
-    candidate = models.ForeignKey('endorsements.Candidate', null=True, blank=True, on_delete=models.SET_NULL)
+    result_choices = (
+        ('win', 'Win'),
+        ('loss', 'Loss'),
+    )
+    result_max_length = 16
+
+    body = RichTextField(verbose_name="Bio")
+    candidate = models.ForeignKey(
+        'endorsements.Candidate',
+        help_text='Ignore - legacy field for old pages.',
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL
+    )
+    donate_url = models.URLField(null=True, blank=True)
     election = models.ForeignKey(
         'endorsements.Election',
         null=True,
         blank=True,
         on_delete=models.SET_NULL
     )
-    signup_tagline = models.CharField(max_length=128, blank=True, null=True)
-    social_image = models.ForeignKey('wagtailimages.Image', null=True, blank=True, on_delete=models.SET_NULL, related_name='+')
-    parent_page_types = ['pages.CandidateEndorsementIndexPage']
+    facebook_url = models.URLField(null=True, blank=True)
+    general_election_date = models.DateField(null=True, blank=True)
+    general_election_result = models.CharField(
+        max_length=result_max_length,
+        choices=result_choices,
+        null=True,
+        blank=True
+    )
+    instagram_url = models.URLField(null=True, blank=True)
+    office = models.CharField(null=True, blank=True, max_length=128)
+    photo = models.ForeignKey(
+        'wagtailimages.Image',
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        related_name='+'
+    )
+    primary_election_date = models.DateField(null=True, blank=True)
+    primary_election_result = models.CharField(
+        max_length=result_max_length,
+        choices=result_choices,
+        null=True,
+        blank=True
+    )
+    social_image = models.ForeignKey(
+        'wagtailimages.Image',
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        related_name='+'
+    )
+    state_or_territory = models.CharField(null=True, blank=True, max_length=32)
```

Comment:
Can we change this to `USStateField` to avoid retyping states/territories and eliminate spelling errors? We are already using localflavor. https://github.com/myusuf3/django-localflavor-us/blob/master/django_localflavor_us/us_states.py

Reply:
@cjmabry My concern about this is that the state field does not have clear predefined values to choose from. I renamed it to state_or_territory field to be more accurate/flexible, because Puerto Rico and DC are not states. Although this does not even make sense always. When Keith Ellison ran for DNC Chair we listed state as Minnesota, but really it's Location = DNC or United States? Or what about president, should be Location = United States? Or are we going to just list the candidate's home state?

I'm curious to hear your thoughts on this, but this is why I avoided dropdown menu.

Reply:
This should be optional in the case of Ellison's DNS run or president. `USStateField` includes all territories including DC and PR. I think calling state or territory is fine, but the idea is to remove a step of typing state names and eliminate mistakes by switching to predefined elections. This will also help standardize data (this was something David said was really helpful on candidate endorsement applications and might help us auto-populate data here as well). If there's a scenario where state or territory isn't applicable (DNC chair, president) we can leave blank.

In the president's case we should probably add a country field but outside of scope of this ticket.

Reply:
@cjmabry I'm ok with trying to use `USStateField` here. But I don't think it should be blank for edge cases, probably better to just put candidate home state as we have been for DNC/POTUS etc.